### PR TITLE
Improved hass package

### PR DESCRIPTION
**Pull request by NicoHood**
As described on the AUR page I've improved lots of things. I recommend to not push the update script nor the readme into the AUR git. It can reside here, but not on AUR.

Thanks for the great work. I might push this package to [community], but first all dependencies and hass itself need to gpg sign their source.

### CODE REVIEW

File: PKGBUILD [hunk 1]
```diff
@@ -1,42 +1,48 @@
...
+#validpgpkeys=('') # TODO https://github.com/home-assistant/home-assistant/issues/9487
 install='hass.install'
 
 prepare() {
-  cd ${srcdir}/${pkgname}-${pkgver}
-  set -e
+  cd "${srcdir}/${pkgname}-${pkgver}"
 
   # typing package is a backport of standard library < 3.5
+  # TODO remove in future versions https://github.com/home-assistant/home-assistant/issues/9525
```
Comment (kamaradclimber) [inline PKGBUILD]: this comment should be one line above

File: PKGBUILD [hunk 2]
```diff
@@ -1,42 +1,48 @@
...
 makedepends=('python-setuptools')
 # NB: this package will install additional python packages in /var/lib/hass/lib depending on components present in the configuration files.
-depends=('python>=3.4' 'python-pip' 'python-requests>=2.14.2' 'python-yaml' 'python-pytz>=2017.2' 'python-vincenty' 'python-jinja>=2' 'python-voluptuous>=0.9.3' 'python-netifaces' 'python-webcolors' 'python-async-timeout>=1.3.0' 'python-aiohttp>=2.2.5' 'python-jinja>=2.9.5' 'python-yarl>=0.11.0' 'python-chardet>=3.0.4' 'python-astral')
+depends=('python' 'python-pip' 'python-requests>=2.14.2' 'python-yaml' 'python-pytz>=2017.2'
+         'python-vincenty' 'python-jinja>=2' 'python-voluptuous>=0.9.3' 'python-netifaces'
+         'python-webcolors' 'python-async-timeout>=1.3.0' 'python-aiohttp>=2.2.5'
+         'python-jinja>=2.9.5' 'python-yarl>=0.11.0' 'python-chardet>=3.0.4' 'python-astral')
 optdepends=('git: install component requirements from github'
             'net-tools: necessary for nmap discovery')
-conflicts=('python-home-assistant' 'python-home-assistant-git')
```
Comment (kamaradclimber) [inline PKGBUILD:16]: why do you remove this line?

File: PKGBUILD [hunk 3]
```diff
@@ -1,42 +1,48 @@
...
+license=('APACHE')
 arch=('any')
 replaces=('python-home-assistant')
 makedepends=('python-setuptools')
 # NB: this package will install additional python packages in /var/lib/hass/lib depending on components present in the configuration files.
-depends=('python>=3.4' 'python-pip' 'python-requests>=2.14.2' 'python-yaml' 'python-pytz>=2017.2' 'python-vincenty' 'python-jinja>=2' 'python-voluptuous>=0.9.3' 'python-netifaces' 'python-webcolors' 'python-async-timeout>=1.3.0' 'python-aiohttp>=2.2.5' 'python-jinja>=2.9.5' 'python-yarl>=0.11.0' 'python-chardet>=3.0.4' 'python-astral')
+depends=('python' 'python-pip' 'python-requests>=2.14.2' 'python-yaml' 'python-pytz>=2017.2'
+         'python-vincenty' 'python-jinja>=2' 'python-voluptuous>=0.9.3' 'python-netifaces'
+         'python-webcolors' 'python-async-timeout>=1.3.0' 'python-aiohttp>=2.2.5'
+         'python-jinja>=2.9.5' 'python-yarl>=0.11.0' 'python-chardet>=3.0.4' 'python-astral')
```
Comment (kamaradclimber) [inline PKGBUILD]: OPT: since you edit this line, could you remove the duplicate python-jinja? thanks

File: home-assistant-tmpfile.conf [hunk 4]
```diff
@@ -0,0 +1 @@
+d /var/lib/hass 0755 hass hass -
```
Comment (kamaradclimber) [inline home-assistant-tmpfile.conf:1]: /var/lib/hass is not a place for tmp files only, configuration files are also stored in this directory. Are you sure this is the right way to declare this kind of directory ?

File: hass.install [hunk 5]
```diff
@@ -1,28 +1,3 @@
...
-  post_install
-  systemctl daemon-reload
-  if [[ "$old_version" < "0.24" ]]; then # TODO remove this code in while (in Sept 2016 for instance)
-    echo "Will migrate database to new format"
-    echo "stopping home-assistant if necessary"
-    systemctl is-enabled home-assistant.service > /dev/null && systemctl stop home-assistant.service
-    echo "Running: hass --script db_migrator --config /var/lib/hass/"
-    sudo -u ${HASS_USER} hass --script db_migrator --config /var/lib/hass/
-    systemctl start home-assistant.service
-  else
```
Comment (kamaradclimber) [inline hass.install:20]: you are deleting the mechanism to restart home assistant during package upgrade. Is this intended?
This, at least, will surprise existing users

File: PKGBUILD [hunk 6]
```diff
@@ -1,42 +1,48 @@
 # Maintainer: Grégoire Seux <grego_aur@familleseux.net>
 # Contributor: Dean Galvin <deangalvin3@gmail.com>
+# Contributor: NicoHood <archlinux {cat} nicohood {dog} de>
+
 pkgname="home-assistant"
-pkgdesc='Home Assistant is an open-source home automation platform running on Python 3'
+pkgdesc='Open-source home automation platform running on Python 3'
 pkgver=0.53.1
-pkgrel=1
+pkgrel=3
```
Comment (kamaradclimber) [inline PKGBUILD:8]: mismatch with .SRCINFO file

**Review by kamaradclimber — CHANGES_REQUESTED** [review]
Thanks for the patch, I've written a few comments and questions.

Regarding committing upgrade script and README, is there a specific reason why you recommend not pushing them to the AUR ? upgrade script is really convenient to manage bi-weekly update of home-assistant. Having two different git repositories (github and aur) feels weird to me.

File: PKGBUILD (hunk 6 above)
Comment (NicoHood) [reply PKGBUILD:8]: I forgot to update. Sorry.

File: PKGBUILD (hunk 3 above)
Comment (NicoHood) [reply PKGBUILD]: Will do.

File: PKGBUILD (hunk 2 above)
Comment (NicoHood) [reply PKGBUILD:16]: Becuse the git packages need to conflict against the main package. And python-home-assistant seems to be obsolete? In this case you would rather use `replace`. But almost everybody should have upgraded by now, and this also only works in the official repository, not the AUR. So conflict is not required at all.

File: PKGBUILD (hunk 1 above)
Comment (NicoHood) [reply PKGBUILD]: will do

File: home-assistant-tmpfile.conf (hunk 4 above)
Comment (NicoHood) [reply home-assistant-tmpfile.conf:1]: I had a quick chat with the other TUs on how to solve this directory issue with user permissions, and that seems to be the way to do it. It is pretty nice and should work, even though the name is misleading.

File: hass.install (hunk 5 above)
Comment (NicoHood) [reply hass.install:20]: This is "not the arch way". You normally do not start or stop services, not even in upgrade scripts. All running services should still continue running, even after an upgrade. You'd have to manually restart them, this is something one *could* add to the upgrade note, but archlinux users should be generally aware of this. 

In general you do not want to have scripts in your package that mess up your system or change configuration.

**Comment by NicoHood** [comment]
You should not upload the readme and update script, as it has nothing to do with the package. The repository is for PKGBUILD stuff, not your personal tooling.

I'd also not advice to use update scripts. The only thing you need to do is to change the pkgver, use updpkgsums and then run extra-x86_64-build. (use a clean chroot, not makepkg directly!). Done.

File: PKGBUILD (hunk 2 above)
Comment (kamaradclimber) [reply PKGBUILD:16]: ok

File: hass.install (hunk 5 above)
Comment (kamaradclimber) [reply hass.install:20]: ok

**Review by kamaradclimber — APPROVED** [review]
thanks for the explaination

**Comment by kamaradclimber** [comment]
thanks for the patch

**Comment by NicoHood** [comment]
Thanks for merging :)

hass also installs missing python dependencies via pip. Those sources are also not gpg signed. I am wondering if we should also list those python packages as optional dependencies, so hass does not need to download all or less of them. What do you think?

Another issue I have is that after the first install and start of the service it will not load. It will fail, as a dependency is missing and after the 2nd restart the service is working. I am not sure if that is an upstream issue or if we could fix it by providing more dependencies. I am not sure yet how to deal with that much hass dependencies. Its quite a lot. Any suggestions?

**Pull request merged**